### PR TITLE
feat: add csrf protection

### PR DESCRIPTION
**Pull request by bhudnell**
Issue #, if available:

Description of changes:

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-app/src/generateRouter.ts [hunk 1]
```diff
@@ -51,6 +52,7 @@ export function generateRouter(apiRouteConfig: ApiRouteConfig): Express {
   // parse application/json
   app.use(express.json());
   app.use(cookieParser());
+  app.use(csurf('none'));
```
Comment (kpark277) [inline solutions/swb-app/src/generateRouter.ts:55]: Is `'none'` necessary? Based on workbench-core/authentication/docs/authenticationMiddleware.md, would passing in no arguments lead to default behavior?

Comment (kpark277) [reply solutions/swb-app/src/generateRouter.ts:55]: Nevermind, I got the the part that default is `'strict'`.

**Review by kpark277 — DISMISSED** [review]

File: workbench-core/authentication/src/authenticationMiddleware.test.ts [hunk 2]
```diff
@@ -35,6 +38,8 @@ const defaultCookieOpts = {
   sameSite: 'strict'
 } as const;
 
+const csrfToken = 'csrfToken';
+
 describe('authenticationMiddleware integration tests', () => {
```
Comment (DaveMadsen) [inline workbench-core/authentication/src/authenticationMiddleware.test.ts:43]: Do you have a test that verifies expected behavior if a csurf token is expected but not included in the request?

**Review by DaveMadsen — COMMENTED** [review]
Please follow up on the testing. Specifically, I didn't see a test for expected behavior when a request is received without the csrf token.

File: workbench-core/authentication/src/authenticationMiddleware.test.ts (hunk 2 above)
Comment (bhudnell) [inline workbench-core/authentication/src/authenticationMiddleware.test.ts:43]:
```suggestion
describe('authenticationMiddleware tests', () => {
```

Comment (bhudnell) [reply workbench-core/authentication/src/authenticationMiddleware.test.ts]: That test should exist but is better suited for integration testing within `workbench-core/example` once that gets implemented.

**Comment by bhudnell** [comment]
Negative tests for CSRF protection should exist but are better suited as integration tests within workbench-core/example once that package gets implemented.

**Review by DaveMadsen — APPROVED** [review]

**Review by kpark277 — APPROVED** [review]

**Pull request merged**